File: C++/Programming Basics with C++/09. Exercise - While-loop/simple-tasks/Vacation.cpp
#include<iostream>
#include <string>

using namespace std;

int main() {
    double vacationPrice, cash;
    cin >> vacationPrice >> cash;

    int spendMoneyCounter = 0;
    int daysCounter = 0;

    while (vacationPrice > cash && spendMoneyCounter < 5) {
        string command;
        cin >> command;
        double money;
        cin >> money;
        daysCounter++;
        if (command == "save") {
            cash += money;
            spendMoneyCounter = 0;
        } else if (command == "spend") {
            cash -= money;
            if (cash < 0) {
                cash = 0;
            }
            spendMoneyCounter++;
            if (spendMoneyCounter >= 5) {
                break;
            }
        }
    }
    if (spendMoneyCounter >= 5) {
        cout << "You can't save the money." << endl;
        cout << daysCounter << endl;
    }
    if (cash >= vacationPrice) {
        cout << "You saved the money for " << daysCounter << " days.";
    }

    return 0;
}

File: C#/Programming Basics with C#/09. While-Loop - Exercise/04. Vacation/Vacation.cs
using System;

class Vacation
{
    static void Main()
    {
        double needMoney = double.Parse(Console.ReadLine());
        double moneyOnhand = double.Parse(Console.ReadLine());
        string choice = "different";
        double amound = 0;
        int spendCounter = 0;
        int daysCounter = 0;

        while (needMoney > moneyOnhand && spendCounter < 5)
        {
            choice = Console.ReadLine();
            amound = double.Parse(Console.ReadLine());
            if (choice == "save")
            {
                moneyOnhand = moneyOnhand + amound;
                spendCounter = 0;
            }
            else if (moneyOnhand > amound && choice == "spend")
            {
                moneyOnhand = moneyOnhand - amound;
                ++spendCounter;
            }
            else if (choice == "spend" && moneyOnhand <= amound)
            {
                moneyOnhand = 0;
                ++spendCounter;
            }
            ++daysCounter;
        }
        if (spendCounter == 5)
        {
            Console.WriteLine("You can't save the money.");
            Console.WriteLine(daysCounter);
        }
        else if (needMoney <= moneyOnhand)
        {
            Console.WriteLine($"You saved the money for {daysCounter} days.");
        }
    }
}
